lib/sales_engine.rb

```diff
@@ -26,14 +26,42 @@ class SalesEngine
 
 
   def initialize(csv_data)
-    @merchants = MerchantRepo.new(csv_data[:merchants], self)
-    @items = ItemRepo.new(csv_data[:items], self)
-    @invoices = InvoiceRepo.new(csv_data[:invoices], self)
-    @invoice_items = InvoiceItemRepo.new(csv_data[:invoice_items], self)
-    @transactions = TransactionRepo.new(csv_data[:transactions], self)
-    @customers = CustomerRepo.new(csv_data[:customers], self)
+    routes1(csv_data)
```

Comment:
Great job at figuring out how to keep the inheritance while getting our tests to pass. This was one of the most difficult things I think to understand but you were able to pull it off! It also makes our initialize method look a lot cleaner.